src/web/app/components/copy-course-modal/copy-course-modal.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { StatusMessageService } from '../../../services/status-message.service';
+import { TimezoneService } from '../../../services/timezone.service';
+import { Course, FeedbackSession } from '../../../types/api-output';
+import { COURSE_ID_MAX_LENGTH } from '../../../types/field-validator';
+
+interface Timezone {
+  id: string;
+  offset: string;
+}
+
+/**
+ * Copy course modal.
+ */
+@Component({
+  selector: 'tm-copy-course-modal',
+  templateUrl: './copy-course-modal.component.html',
+  styleUrls: ['./copy-course-modal.component.scss'],
+})
+export class CopyCourseModalComponent implements OnInit {
+
+  // const
+  COURSE_ID_MAX_LENGTH: number = COURSE_ID_MAX_LENGTH;
+
+  @Input()
+  courseToFeedbackSession: Record<string, FeedbackSession[]> = {};
+
+  @Input()
+  activeCourses: Course[] = [];
+
+  @Input()
+  allCourses: Course[] = [];
+
+  isCopyFromOtherSession: boolean = false;
+  hasCourseId: boolean = false;
+  timezones: Timezone[] = [];
+  newTimezone: string = '';
+  newCourseId: string = '';
+  newCourseName: string = '';
+  oldCourseId: string = '';
+  oldCourseName: string = '';
+
+  selectedFeedbackSessions: Set<FeedbackSession> = new Set<FeedbackSession>();
+
+  constructor(public activeModal: NgbActiveModal,
+              private statusMessageService: StatusMessageService,
+              private timezoneService: TimezoneService) {}
+
+  ngOnInit(): void {
+    Object.entries(this.timezoneService.getTzOffsets())
+      .map(([id, offset]: [string, number]) => {
```

Comment:
When I mentioned using `.map`, i meant something like `this.timezones = Object.entries(this.timezoneService.getTzOffsets()).map(...)`, so that you don't have to use `.push()`